### PR TITLE
OdraType and OdraError derive macros

### LINKED ISSUE #232: OdraError
Errors are numbers.

Users should be able to define their own errors. Possible code:
```rust
pub enum Erc20Error {
    InsufficientBalance = 1,
    InsufficientAllowance = 2
}

impl From<Erc20Error> for OdraError {
    fn from(error: Erc20Error) -> Self {
        OdraError::user(error as u32)
    }
}
```

Users should be able to easily test errors.
```rust
assert_eq!(erc20.try_transfer(bob, 10.into()), Err(Erc20Error::InsufficientBalance));
```

Error code spaces similar to existing impl.

### LINKED ISSUE #236: `OdraType` macro.
It should derive `ToBytes`, `FromBytes` and `CLTyped`.

### PR DESCRIPTION
**Pull request by kpob**
OdraType derive macro generates
ToBytes, FromBytes, CLTyped, Clone

OdraError macro generates
From<#error> for odra::OdraError

### CODE REVIEW

File: core/src/variable.rs [hunk 1]
```diff
@@ -43,6 +38,13 @@ impl<T: FromBytes + Default> Variable<T> {
     }
 }
 
+impl<T: FromBytes + Default> Variable<T> {
+    pub fn get_or_default(&self) -> T {
+        let env = self.env();
+        env.get_value(&env.current_key()).unwrap_or_default()
```
Comment (zie1ony) [inline core/src/variable.rs]:
```suggestion
self.get().unwrap_or_default()
```

File: examples2/src/counter.rs [hunk 2]
```diff
@@ -1,100 +1,80 @@
 use odra::prelude::*;
-use odra::{runtime_args, FromBytes, RuntimeArgs};
+use odra::{runtime_args, Address, FromBytes, OdraType, RuntimeArgs};
```
Comment (zie1ony) [inline examples2/src/counter.rs]: Could you remove `runtime_args`, `FromBytes` and `RuntimeArgs`?

File: modules/src/erc20.rs [hunk 3]
```diff
@@ -1,7 +1,7 @@
-use odra::prelude::*;
-use odra::{Address, ContractEnv, Mapping, U256, Variable};
 use crate::erc20::errors::Error::{DecimalsNotSet, NameNotSet, SymbolNotSet};
 use crate::erc20::events::*;
+use odra::prelude::*;
+use odra::{Address, ContractEnv, Mapping, Variable, U256};
```
Comment (zie1ony) [inline modules/src/erc20.rs:4]: Can you remove `ContractEnv`?

File: core/src/variable.rs [hunk 4]
```diff
@@ -43,6 +38,13 @@ impl<T: FromBytes + Default> Variable<T> {
     }
 }
 
+impl<T: FromBytes + Default> Variable<T> {
```
Comment (kubaplas) [inline core/src/variable.rs:40]: :+1:

**Review by kubaplas — APPROVED** [review]

**Pull request merged**